device_backends/LogicalNameMapping/src/LNMBitRangeAccessPlugin.cc

```diff
@@ -99,32 +101,26 @@ namespace ChimeraTK::LNMBackend {
       auto unlock = cppext::finally([this] { this->_lock.unlock(); });
       _target->postRead(type, hasNewData);
       if(!hasNewData) return;
-      auto validity = _target->dataValidity();
 
-      if constexpr(!std::is_integral<UserType>::value) {
-        // This should never be reached, the decorate function should make sure that we only have integral types here
-        assert(false);
-      }
-      else {
-        uint64_t v{};
-        static_assert(sizeof(TargetType) <= sizeof(uint64_t), "Target data type too big.");
-
-        auto targetValue = _target->accessData(0);
-        memcpy(std::addressof(v), std::addressof(targetValue), sizeof(TargetType));
+      if constexpr(std::is_same_v<uint64_t, TargetType>) {
+        auto validity = _target->dataValidity();
+        uint64_t v{_target->accessData(0)};
         v = (v & _maskOnTarget) >> _shift;
 
+        buffer_2D[0][0] = fixedPointConverter.scalarToCooked<UserType>(uint32_t(v));
         // There are bits set outside of the range of the UserType
         // Clamping according to B.2.4 and setting the faulty flag
```

Comment:
Shouldn't this be made part of the fixed point converter in the end as well? Please add a comment and the ticket number.

Reply:
Changed the comment, that was wrong anyway (the code did not clamp anymore)